### PR TITLE
Fix and refactor X4 Foundations module

### PR DESCRIPTION
**Pull request by SuperFlue**
This module was using the wrong directory for mods so i fixed it, and changed some other stuff since i was working on it anyway.

- Replaced old detection method with detection using Steam ID. 
- Changed mod directory to extensions folder in game root, as using the one in Documents breaks a lot of mods. (Example [here](https://www.nexusmods.com/x4foundations/mods/285), [here](https://www.nexusmods.com/x4foundations/mods/305) and [here](https://www.nexusmods.com/x4foundations/mods/4) )
- Changed names of test and install content functions to be more explicit.
- Some refactoring of elements and general tidying of formatting (removing a bunch of newlines).

### CODE REVIEW

**Comment by SuperFlue** [comment]
Actually having a second look at the commit history, it seems like commit ``e9f7d80`` broke mod support, it was correct before that....

**Comment by IDCs** [comment]
Hi @SuperFlue, can you please explain the difference between the documents and extensions mod paths and why certain mods function within one folder and not within the other. 

The installation pattern has been changed to the documents folder as we were led to believe by the X4 modding community that most mods will not function correctly inside the extensions folder.

**Comment by SuperFlue** [comment]
Hi @IDCs, just had to double check my findings.

As a baseline I first cleared out all mods to remove the "modified" tag that the game shows in the main menu (just to not have any other mods interfering).

After that I placed the Sectorsatellites mod in the extensions folder in the game root. Starting the game shows "modified" in the main menu and after starting a new game I get a message that the mod has successfully been enabled (this message is created by the mod itself which I will get back to in a bit).

Moving the mod to the extensions folder in documents and launching the game gives the modified tag (mod is detected), but when starting a new game I get a message that assets has not been found and loading the mod failed.

(Note that there are basically two types of mods, one is just loose files while the other is packing into catalog archives. Sectorsatellites is a packaged mod.)

Now it could be that the mod was doing a check to see if it was placed in documents. So I extracted the files from the catalog file to investigate the scripts.
The mod trows the missing assets error when it is in fact unable to find its assets.
Basically it is trying to find a file with the path ``extensions\sectorsatellites\assets\props\equipment\satelite\macros\advanced_satellite_detection_probe_macro``.
When mod is started from game root this returns successfully while started from documents it fails.

Also I should note that Steam Workshop mods are also placed in the game root extensions folder, which I would take as the official word that this is indeed the folder to use (Steam mods are also packed in catalogs).

Side note: The current Vortex module also placed the extensions folder in "Documents\Egosoft\X4\" instead of "Documents\Egosoft\X4\<Steam UID>\" which caused the game not to see the mod at all.

**Comment by SuperFlue** [comment]
I see that you have made some modifications to the module, so i guess this PR is no longer entirely valid....
Tough i will admit i do not entirely understand the change in the QueryModPath function.
It looks like if the game has been discovered in Steam you use the documents folder but if it the GOG version or path is manually selected game root is used?

**Comment by IDCs** [comment]
Hi @SuperFlue, apologies for the late reply; the PR should still be ok as long as I resolve the conflicts before merging so no worries on that front. The QueryModPath changes were made to add support for the GoG edition of the game. Basically what it does is, if we find a steam entry for the game, we install mods inside the documents folder as we've been doing so far; otherwise we assume that the user has a GoG copy so we deploy to the extensions folder inside the game's root directory.

So what you're saying is that packaged mods are to be extracted inside the extensions directory; while the loose file mods _should_ be ok to extract to documents ?

I'm going to enhance the extension and add 2 modTypes with different destination folders. The issue I'm having with this is our inability to correctly identify which modType a mod belongs to programatically. Given this fact I think I'm going to change the extension so that all mods will be assigned to the "extensions" directory by default, and we will allow users to manually change modType to extract into the documents folder if needed.

**Comment by IDCs** [comment]
Hi @SuperFlue, unfortunately due to the extent of the changes that were required to the extension as a whole, I had opted not to merge your PR, but have modified it according to your guidelines.

The commits for this are:
https://github.com/Nexus-Mods/vortex-games/commit/96380be2c483e00f9264af1f0823de76b5a14e79
https://github.com/Nexus-Mods/vortex-games/commit/b76cbd76cf92a68a1badb84ecee6070d33e0f709

Let me know if you experience any further issues with it.

**Comment by SuperFlue** [comment]
Hi @IDCs I did forget to properly test if the unpacked mod would load from Documents.

I tested now and it seems like the game will not load anything from the "Assets" sub-directory of a mod if loaded from Documents.
I would recommend completely removing the usage of Documents for X4 mods. Since it's for all intents and purposes not fully supported as a modding directory, and there is no indication of it changing at this point.
Tough if it does change, I expect the developers would indicate that in a patch changelog.

Outside of that, the committed changes are good, thanks! 👍
